Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/passthroughproject/PassThroughProjectManager.java

```diff
@@ -0,0 +1,141 @@
+import edu.asu.spring.quadriga.dto.PassThroughProjectDTO;
+import edu.asu.spring.quadriga.exceptions.DocumentParserException;
+import edu.asu.spring.quadriga.exceptions.NoSuchRoleException;
+import edu.asu.spring.quadriga.exceptions.QStoreStorageException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.passthroughproject.constants.Constants;
+import edu.asu.spring.quadriga.service.impl.BaseManager;
+import edu.asu.spring.quadriga.service.network.INetworkManager;
+import edu.asu.spring.quadriga.service.passthroughproject.IPassThroughProjectDocumentReader;
+import edu.asu.spring.quadriga.service.passthroughproject.IPassThroughProjectManager;
+import edu.asu.spring.quadriga.service.workspace.IExternalWSManager;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+@Service
+public class PassThroughProjectManager extends BaseManager implements IPassThroughProjectManager {
+
+    @Autowired
+    private IWorkspaceDAO workspaceDao;
+
+    @Autowired
+    private INetworkManager networkManager;
+
+    @Autowired
+    private IExternalWSManager externalWSManager;
+
+    @Autowired
+    private IPassThroughProjectDAO projectDao;
+
+    @Autowired
+    private IProjectSecurityChecker projectSecurityChecker;
+
+    @Resource(name = "projectconstants")
+    private Properties messages;
+
+    @Autowired
+    private IPassThroughProjectDocumentReader passThroughProjectDocumentReader;
+
+    @Override
+    @Transactional
+    public String addPassThroughProject(String userid, IPassThroughProject project) throws QuadrigaStorageException {
+        return projectDao.addPassThroughProject(userid, project);
+    }
+
+    @Override
+    @Transactional
+    public String getInternalProjectId(String externalProjectid, String userid)
+            throws QuadrigaStorageException, NoSuchRoleException {
+
+        List<PassThroughProjectDTO> projectDTOs = projectDao.getExternalProjects(externalProjectid);
+
+        for (PassThroughProjectDTO projectDTO : projectDTOs) {
+            boolean isOwner = projectSecurityChecker.isProjectOwner(userid, projectDTO.getProjectid());
+            if (isOwner) {
+                return projectDTO.getProjectid();
+            }
+            boolean isCollaborator = projectSecurityChecker.isCollaborator(userid,
+                    RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN, projectDTO.getProjectid());
+            if (isCollaborator) {
+                return projectDTO.getProjectid();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String callQStore(String xml, IUser user) throws QStoreStorageException, DocumentParserException {
+        String networkName = Constants.VOGONWEB_NETWORK_NAME;
+
+        String userid = user.getUserName();
+        String networkId = null;
+        try {
+            Document document = passThroughProjectDocumentReader.getXMLParser(xml);
```

Comment:
don't pass jdom document up to this class. Manager shouldn't know that jdom is used.